policy/pa.go

```diff
@@ -49,64 +50,125 @@ func New(challengeTypes map[string]bool) (*AuthorityImpl, error) {
 	return &pa, nil
 }
 
-type blacklistJSON struct {
-	Blacklist      []string
-	ExactBlacklist []string
+// blockedNamesPolicy is a struct holding lists of blocked domain names. One for
+// exact blocks and one for blocks including all subdomains.
+type blockedNamesPolicy struct {
+	// ExactBlockedNames is a list of domain names. Issuance for names exactly
+	// matching an entry in the list will be forbidden. (e.g. `ExactBlockedNames`
+	// containing `www.example.com` will not block `example.com` or
+	// `mail.example.com`).
+	//
+	// TODO(@cpu): Remove the JSON tag when data is updated to use the new field names
+	ExactBlockedNames []string `json:"exactBlacklist" yaml:"ExactBlockedNames"`
+	// BlockedNames is like ExactBlockedNames except that issuance is blocked for
+	// subdomains as well. (e.g. BlockedNames containing `example.com` will block
+	// `www.example.com`).
+	//
+	// This list typically holds high-value domain names and doesn't change with
+	// much regularity.
+	//
+	// TODO(@cpu): Remove the JSON tag when data is updated to use the new field names
+	BlockedNames []string `json:"blacklist" yaml:"BlockedNames"`
+
+	// SDNNames operate the same as BlockedNames but change with greater
+	// frequency based on the Treasury department's Specially Designated Nationals
+	// list. Because of the rate of change and the importance of careful
+	// management of these names we separate them from the BlockedNames list.
+	SDNNames []string `yaml:"SDNNames"`
```

Comment:
At the risk of bike-shedding myself I think a better field name might be `RevokedNames` since it's somewhat more general than the SDN case specifically.

Reply:
I slept on it and think that `RevokedNames` is better: 6c323b6